crates/store/re_chunk_store/src/dataframe.rs

```diff
@@ -0,0 +1,499 @@
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for ComponentColumnDescriptor {
+    #[inline]
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        let Self {
+            entity_path,
+            archetype_name,
+            archetype_field_name,
+            component_name,
+            datatype: _,
+            is_static: _,
+        } = self;
+
+        entity_path
+            .cmp(&other.entity_path)
+            .then_with(|| component_name.cmp(&other.component_name))
+            .then_with(|| archetype_name.cmp(&other.archetype_name))
+            .then_with(|| archetype_field_name.cmp(&other.archetype_field_name))
+    }
+}
+
+impl std::fmt::Display for ComponentColumnDescriptor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let Self {
+            entity_path,
+            archetype_name,
+            archetype_field_name,
+            component_name,
+            datatype: _,
+            is_static,
+        } = self;
+
+        let s = match (archetype_name, component_name, archetype_field_name) {
+            (None, component_name, None) => component_name.to_string(),
+            (Some(archetype_name), component_name, None) => format!(
+                "{entity_path}@{}::{}",
+                archetype_name.short_name(),
+                component_name.short_name(),
+            ),
+            (None, component_name, Some(archetype_field_name)) => format!(
+                "{entity_path}@{}#{archetype_field_name}",
+                component_name.short_name(),
+            ),
+            (Some(archetype_name), component_name, Some(archetype_field_name)) => format!(
+                "{entity_path}@{}::{}#{archetype_field_name}",
+                archetype_name.short_name(),
+                component_name.short_name(),
+            ),
+        };
+
+        if *is_static {
+            f.write_fmt(format_args!("|{s}|"))
+        } else {
+            f.write_str(&s)
+        }
+    }
+}
+
+impl ComponentColumnDescriptor {
+    #[inline]
+    pub fn new<C: re_types_core::Component>(entity_path: EntityPath) -> Self {
+        Self {
+            entity_path,
+            archetype_name: None,
+            archetype_field_name: None,
+            component_name: C::name(),
+            datatype: C::arrow_datatype(),
+            // TODO(cmc): one of the many reasons why using `ComponentColumnDescriptor` for this
+            // gets a bit weird… Good enough for now though.
+            is_static: false,
+        }
+    }
+
+    #[inline]
+    pub fn to_arrow_field(&self) -> ArrowField {
+        let Self {
+            entity_path,
+            archetype_name,
+            archetype_field_name,
+            component_name,
+            datatype,
+            is_static,
+        } = self;
+
+        ArrowField::new(
+            component_name.short_name().to_owned(),
+            datatype.clone(),
+            false, /* nullable */
+        )
```

Comment:
In order to successfully build an arrow2 RecordBatch using the produced schema, I had to make the following change:
```suggestion
        ArrowField::new(
            component_name.short_name().to_owned(),
            ArrowDatatype::List(std::sync::Arc::new(ArrowField::new(
                "item",
                datatype.clone(),
                true, /* is_nullable = true; This seems backwards to me. */
            ))),
            false, /* is_nullable = true; This seems backwards to me. */
        )
```

However, this change appears to require nullabillity flags that are backwards from my intuition.  I would have expected the INNER list to be non-nullable -- we can't null individual elements within a batch. But the OUTER list to be nullable -- individual rows can be null.

Reply:
The way you've set those nullability flags makes sense to me:
* for a list datatype, the nullability flag refers to its items.
* for a top-level field, the nullability datatype should logically describe whether the entire field can be missing from the recordbatch payload -- though from what ive seen in practice it mostly just means gibberish and you have to set it to whatever value makes the system you're working with happy.

The fact that the list layer is missing at all is a very real problem though >< I forgot that `TransportChunk` made that one automagically work :sparkles: